resources/function_help/json/array_replace

```diff
@@ -0,0 +1,10 @@
+{
+  "name": "array_replace",
```

Comment:
I wasn't aware of this function and I think it would make sense to extend it.
I see only one little negative point: the "replace" function is in the "String" group. Probably someone working with arrays will not search this function there.

P.S. All this could also apply to the function "length" which have his counterpart "array_length". But this could broke existing expressions... I don't know since how long is "array_length" existing
P.P.S Maybe this function should be in the group "General" instead?

Reply:
A function can be displayed in multiple groups. see e.g. length (both string and geometry)

Reply:
Oh nice than I'll do same way thanks!

Reply:
Hmmm I know I'm the one who suggested that but on a second thought, I wonder if it was a so good idea. Having 4 variants for a function leads to a huge help panel and I'm afraid it might look confusing to the end users (I forgot there was a map variant). And keeping the array_replace maybe makes the function more identifiable and inline with the other array functions? 
Others, opinion? And sorry for the extra work (though we should keep ability to replace multiple values in a row).

Reply:
I agree that it looks cleaner with separated functions. So I re-splitted the functions and added to array_replace the ability to replace by array or by map.
Don't worry for the extra work, it is still learning time for me so it is good look around a little in other corners ;)